### PR TITLE
TagBox: propagate maxFilterQueryLength option to a selection (T1191760)

### PR DESCRIPTION
**Pull request by iBat**

### CODE REVIEW

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js [hunk 1]
```diff
@@ -20,6 +20,7 @@ import ODataStore from 'data/odata/store';
 import TagBox from 'ui/tag_box';
 import { normalizeKeyName } from 'events/utils/index';
 import { getWidth, getHeight } from 'core/utils/size';
+import guid from 'core/guid';
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]: it's a class as far as I remember. Let's use code style consistent with other places
```suggestion
import Guid from 'core/guid';
```

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js [hunk 2]
```diff
@@ -7463,6 +7464,68 @@ QUnit.module('regression', {
 
         assert.ok(true, 'Widget rendered');
     });
+
+    QUnit.test('option maxFilterQueryLength should be propagated to selection controls and let exceed its default limit (T1191760)', function(assert) {
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]:
```suggestion
    QUnit.test('maxFilterQueryLength option should be propagated to selection and allow to exceed exceed its default limit (T1191760)', function(assert) {
```

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js [hunk 3]
```diff
@@ -7463,6 +7464,68 @@ QUnit.module('regression', {
...
+                maxFilterQueryLength: 9999,
+                deferRendering: false,
+            });
+
+        const $inputWrapper = $element.find(`.${DROP_DOWN_EDITOR_INPUT_WRAPPER}`);
+
+        this.clock.tick(50);
+        $inputWrapper.trigger('dxclick');
+
+        const selectAllCheck = $(`.${SELECT_ALL_CHECKBOX_CLASS}`).dxCheckBox('instance');
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]:
```suggestion
        const selectAllCheckBox = $(`.${SELECT_ALL_CHECKBOX_CLASS}`).dxCheckBox('instance');
```

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js [hunk 4]
```diff
@@ -7463,6 +7464,68 @@ QUnit.module('regression', {
...
+
+        assert.equal(selectAllCheck.option('value'), true, 'the "select all" checkbox is checked');
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]:
```suggestion
        assert.strictEqual(selectAllCheck.option('value'), true, 'the "select all" checkbox is checked');
```

**Review by ksercs — APPROVED** [review]

**Review by alexanderPolosatov — APPROVED** [review]

**Pull request merged**